rts/motoko-rts/src/stabilization/layout/stable_object.rs

```diff
@@ -58,6 +58,16 @@ impl StaticScanner<Value> for Object {
     }
 }
 
+const NON_STABLE_OBJECT_TAGS: [Tag; 1] = [TAG_CLOSURE];
+
+fn has_non_stable_type(old_field: Value) -> bool {
+    unsafe { old_field.is_ptr() && NON_STABLE_OBJECT_TAGS.contains(&old_field.tag()) }
+}
+
+// Dummy value used for non-stable fields are potentially contained in stable records because of structural sub-typing.
+// Must be a non-skewed value such that the GC also ignores this value.
+const DUMMY_VALUE: StableValue = StableValue(0);
+
```

Comment:
I don't think this is sufficient to handle all closures that could occur in stable values (see PR comment https://github.com/dfinity/motoko/pull/4286#discussion_r1404674463

Reply:
Yes, this is indeed insufficient. I fixed it. Thanks a lot for noticing and sorry for the error.